### PR TITLE
docs: update RELEASE_CHECKLIST.md

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Update RELEASE_CHECKLIST.md per suggestions from [Notary Project community meeting](https://hackmd.io/_vrqBGAOSUC_VWvFzWruZw?view#Notes) and https://github.com/notaryproject/.github/issues/28. Please note that this PR focuses on updating the Notation CLI release process only. 

I will create another PR to add the release checklist for releasing a patch.

### CODE REVIEW

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:16]: What's the benefit of creating an annotated tag instead of a lightweighted tag where the tag message is exactly the same as the tag name?

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [reply RELEASE_CHECKLIST.md:17]: Will you consider just push the specific newly created tag?

File: RELEASE_CHECKLIST.md
Comment (priteshbandi) [inline RELEASE_CHECKLIST.md:12]: There are 3 different `go.mod` file in notation pkg. Can we please list all the files for each package where changes needs to made and better if we can write commands?

File: RELEASE_CHECKLIST.md
Comment (priteshbandi) [inline RELEASE_CHECKLIST.md:20]: Can we please add specific links?

File: RELEASE_CHECKLIST.md
Comment (yizha1) [inline RELEASE_CHECKLIST.md:16]: We also need to sign the tag based on current rules. So `-s` flag is required.

**Review by yizha1 — COMMENTED** [review]
Thanks @FeynmanZhou

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Changed to use a lightweight tag

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Make sense. Changed to push the specific tag

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Added links for all docs

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Added `-s`

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: @priteshbandi  This checklist is currently used for the Notation CLI repo only. There is another plan to update the release checklist and process document by standing on the whole Notation project's point of view and involve notation-go and notation-core-go projects in a central place. see https://github.com/notaryproject/.github/issues/28. That said we will create a doc in `.github` repo and consolidate the release process for notation, notation-go, notation-core-go, notaryproject(spec).

Is it acceptable?

Comment (yizha1) [reply RELEASE_CHECKLIST.md]: @FeynmanZhou it seems still in `notation`  repo, for example, `go.mod` for e2e testing needs to be updated as well.

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:13]: nit:
```suggestion
1. Create an issue to vote for the new release. Add the link of change logs and determine a specific git commit to tag on the issue description. List all repo-level maintainers and make sure to have majority of approvals from the maintainers before releasing it. 
```

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:9]: super nit:
```suggestion
1. Check if there are any security vulnerabilities fixed and security advisories published before a release. Security advisories should be linked on the release notes.
```

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:12]: Should we mention that the PR is merged with all E2E test cases passed before starting the next step?

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:14]:
```suggestion
1. After the version and dependencies are updated, on the `main` branch of the notation repository (not a fork), execute `git clone https://github.com/notaryproject/notation.git`. 
```

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: @yizha1  Good catch. Add https://github.com/notaryproject/notation/blob/main/test/e2e/go.mod to this step

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:12]: Looks like we could combine these two PRs into one PR, which contains dependency updates plus version bump up?

Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: I agree. Combined and revised these two steps

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:14]: nit: the URL represents that it is the original repository.
```suggestion
1. After the version and dependencies are updated, execute `git clone https://github.com/notaryproject/notation.git` to clone the repository to your local file system. 
```

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:13]: We need two PRs: one for dependency update, another for version update for voting. The PR title can be `bump: tag and release $version`

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:11]: How about the `go.mod` and `go.sum` in `test/e2e/plugin`?

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:13]: Let's merge these 3 steps into one. Note: PR 748 is not for the purpose of updating the dependencies.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:11]: "Update the `Version` in [internal/version/version.go](internal/version/version.go#L5) in the forked repository." <- This sentence should be removed.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:14]: "Create another PR to update the Notation CLI version" with a single commit. The commit message must follow conventional commit and could be `bump: tag and release $version`. Record the digest of that commit as `<commit_digest>`.

Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:14]: Wait for the above PR to be merged before creating this PR.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:15]: This should be done after the above PR is merged.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:14]: This PR MUST be merged using `Create a merge commit` method in GitHub.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:18]: Is that `git push origin $version`?

**Review by shizhMSFT — COMMENTED** [review]
Let's fix one thing at a time. We can leave branching in the next PR.

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:11]: > follow the two steps above and cut the release for the notation library

This sentence confuses me. Releasing `notation` CLI does not mean releasing its dependent libraries.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:15]: nit: `PR 754` --> `PR#754` or `PR #754`

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:11]: Missing a dot at the end.

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:15]: > Run `go mod tidy` locally to check if the `go.sum` file is also updated with any potential changes.

It seems redundant to the previous step

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:17]: After the vote PR being merged is sufficient.

File: RELEASE_CHECKLIST.md
Comment (yizha1) [inline RELEASE_CHECKLIST.md:23]:
```suggestion
1. Update the necessary documentation in the [notaryproject.dev](https://github.com/notaryproject/notaryproject.dev) repository to reflect the changes of the release on the Notary Project website, includes but not limited to [installation guide](https://github.com/notaryproject/notaryproject.dev/blob/main/content/en/docs/installation/cli.md), [user guide](https://github.com/notaryproject/notaryproject.dev/tree/main/content/en/docs/how-to), [banner](https://github.com/notaryproject/notaryproject.dev/blob/main/layouts/partials/banner.html), [release blog](https://github.com/notaryproject/notaryproject.dev/tree/main/content/en/blog).

```

Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (FeynmanZhou) [reply RELEASE_CHECKLIST.md]: Updated

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:17]: Version and dependencies are already updated before voting.
```suggestion
1. After the voting PR is merged in the [Notation](https://github.com/notaryproject/notation.git) repository, execute `git clone https://github.com/notaryproject/notation.git` to clone the repository to your local file system.

```

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:16]: nit:
```suggestion
1. Create another PR to update the Notation CLI version with a single commit when PRs in above steps are merged. The commit message MUST follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) and could be `bump: tag and release $version`. Record the digest of that commit as `<commit_digest>`. This PR is also used for voting purpose of the new release. Add the link of change logs and repo-level maintainer list in the PR's description. The PR title could be `bump: tag and release $version`. Make sure to reach a majority of approvals from the repo-level maintainers before releasing it. This PR should be merged using [Create a merge commit](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) method in GitHub. See [PR #748](https://github.com/notaryproject/notation/pull/748) as an example. 
```

**Comment by iamsamirzon** [comment]
LGTM but IAMNAM

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by shizhMSFT — DISMISSED** [review]
LGTM

File: RELEASE_CHECKLIST.md
Comment (shizhMSFT) [inline RELEASE_CHECKLIST.md:14]: It seems this bulleted list is not well rendered.

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:19]: The tag should be annotated:
```suggestion
- Create a tag by running `git tag -am $version $version -s`.
```

File: RELEASE_CHECKLIST.md
Comment (Two-Hearts) [inline RELEASE_CHECKLIST.md:5]: Since this is a release checklist, under the `Release Process` session, we should add a new section for branch/patch releases, i.e., when there's new features in `main` but we just want a `patch` release.

**Review by JeyJeyGao — APPROVED** [review]
LGTM

File: RELEASE_CHECKLIST.md
Comment (yizha1) [reply RELEASE_CHECKLIST.md:5]: created a new issue for tracking this comment #985

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**
